docs/pages/contracts/v4/guides/perform-a-swap.mdx

```diff
@@ -0,0 +1,475 @@
+}
+```
+
+### PERMIT2_PERMIT
+
+Allow users to approve token allowances in a single transaction.
+
+```solidity
+{
+  IAllowanceTransfer.PermitSingle calldata permitSingle; // see IAllowanceTransfer for struct detail 
+  bytes calldata data; // signature for the permit 
+}
+```
+
+### WRAP_ETH
+
+Convert Ether (ETH) into Wrapped Ether (WETH)
+
+```solidity
+{
+  address recipient; // recipient after conversion 
+  uint256 amount; // revert if amount to wrap is less than amount
+}
+```
+
+### UNWRAP_WETH
+
+Convert Wrapped ETH (WETH) into Ether (ETH)
+
+```solidity
+{
+  address recipient; // recipient after conversion 
+  uint256 amountMin; // revert if ETH amount in contract is less than  amountMin
+}
+```
+
+### PERMIT2_TRANSFER_FROM_BATCH
+Facilitate batch transfer of ERC-20 tokens using the Permit2 smart contract.
+
+```solidity
+{
+  // see IAllowanceTransfer for struct detail 
+  IAllowanceTransfer.AllowanceTransferDetails[] calldata batchDetails;
+}
+```
+
+### BALANCE_CHECK_ERC20
+
+Verify the balance of a specific ERC-20 token held by an address
+
+```solidity
+{
+  address owner; // address to check 
+  address token; // token to check 
+  uint256 minBalance; // revert if token.balanceOf(owner) <= minBalance
+}
+```
+
+### V4_SWAP
+Perform a swap on v4 - see example of performing a swap in v4 section below.
+
+```solidity
+{
+    bytes payload;
+}
+```
+
+### V3_POSITION_MANAGER_PERMIT
+
+Approve an NFT. This should only be used for v3 to v4 migration. If only this command is done, it means others can decrease your v3 NFT liquidity and take your token.
+
+```solidity
+{
+    bytes data; 
+}
+
+data = abi.encodeWithSelector(IERC721Permit.permit.selector, 
+  address spender, uint256 deadline, uint8 v, bytes32 r, bytes32 s);
+
+```
+
+### V3_POSITION_MANAGER_CALL
+
+Make a call to v3 NonFungiblePositionManager. The only methods available to call are decreaseLiquidity, collect and burn.
+
+```solidity
+{
+    bytes data;
+}
+
+// 'data' varies depending on which method of the v3 NonFungiblePositionManager is being called
+
+// example 1: decrease
+IV3NonfungiblePositionManager.DecreaseLiquidityParams memory decreaseParams = 
+    IV3NonfungiblePositionManager.DecreaseLiquidityParams({
+        tokenId: tokenId,
+        liquidity: liqudiity,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: block.timestamp
+    });
+input[i] = abi.encodeWithSelector(IV3NonfungiblePositionManager.decreaseLiquidity.selector, decreaseParams);
+
+// example 2: collect
+IV3NonfungiblePositionManager.CollectParams memory collectParam = 
+    IV3NonfungiblePositionManager.CollectParams({
+        tokenId: v3TokenId,
+        recipient: address(router),
+        amount0Max: type(uint128).max,
+        amount1Max: type(uint128).max
+    });
+input[i] = abi.encodeWithSelector(IV3NonfungiblePositionManager.collect.selector, collectParam);
+```
+
+### V4_CL_POSITION_CALL
```

Comment:
Should we add some warning here ? For example, do not permit or approve this contract over a v4 position or someone could use this command to decrease/burn your position hence should only call `Actions.CL_MINT_POSITION` etc.

Reply:
Or we can consider not including those position related actions because this page is a guide about `Perform a swap in v4`

Reply:
this would be moved to `universal router` section once v4 is live.. so likely the command doc can stay.

<img width="752" alt="Screenshot 2024-09-02 at 11 15 28" src="https://github.com/user-attachments/assets/a40bacfb-3c28-4d9a-8fdd-cd3a835b689b">

will add that warning